omod/src/main/java/org/openmrs/module/owa/filter/OwaFilter.java

```diff
@@ -49,23 +50,31 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
 			requestURL = request.getServletPath();
 		}
 		
+		String loginUrl = Context.getAdministrationService().getGlobalProperty("login.url", "login.htm");
 		if (Context.isAuthenticated()) {
 			if (requestURL.startsWith(owaBasePath)) {
 				String newURL = requestURL.replace(owaBasePath, "/ms/owa/fileServlet");
 				req.getRequestDispatcher(newURL).forward(req, res);
 			} else {
-				chain.doFilter(req, res);
+				if (loginUrl.contains("addonmanager") && (request.getServletPath().equals("/index.htm") || request.getServletPath().equals("/login.htm") || request.getServletPath().equals("/"))) {
+					((HttpServletResponse) res).sendRedirect(request.getContextPath() + "/" + loginUrl);
+				} else {
+					chain.doFilter(req, res);
+				}
 			}
 		} else {
 			if (requestURL.startsWith(owaBasePath)) {
 				String newURL = requestURL.replace(owaBasePath, "/ms/owa/redirectServlet");
-				String loginUrl = Context.getAdministrationService().getGlobalProperty("login.url", "login.htm");
-				if (requestURL.contains(loginUrl)) {
+				if (requestURL.contains(loginUrl) || requestURL.contains("addonmanager")) {
 					newURL = requestURL.replace(owaBasePath, "/ms/owa/fileServlet");
 				}
 				req.getRequestDispatcher(newURL).forward(req, res);
 			} else {
-				chain.doFilter(req, res);
+				if (loginUrl.contains("addonmanager") && (request.getServletPath().equals("/index.htm") || request.getServletPath().equals("/login.htm") || request.getServletPath().equals("/"))) {

```

Comment:
What happens when the loginUrl has a value which is none of the ones you have tested above?

Reply:
Thid means the `login.url` GP property must have been set by an installed module that is ready to handle that login URL.

Reply:
If not, then just like other requests that do not have the above specs, the unique logins are passed/chained on for further filtering towards their destinations.